internal/services/compute/image_data_source_test.go

```diff
@@ -0,0 +1,139 @@
+package compute_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurestack/internal/tf/acceptance"
+	"github.com/hashicorp/terraform-provider-azurestack/internal/tf/acceptance/check"
+)
+
+type ImageDataSource struct{}
+
+func TestImageDataSource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurestack_image", "test")
+	r := ImageDataSource{}
+
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("resource_group_name").Exists(),
+				check.That(data.ResourceName).Key("os_disk.#").HasValue("1"),
+				check.That(data.ResourceName).Key("os_disk.0.blob_uri").Exists(),
+				check.That(data.ResourceName).Key("os_disk.0.caching").HasValue("None"),
+				check.That(data.ResourceName).Key("os_disk.0.os_type").HasValue("Linux"),
+				check.That(data.ResourceName).Key("os_disk.0.os_state").HasValue("Generalized"),
+				check.That(data.ResourceName).Key("os_disk.0.size_gb").HasValue("30"),
+				check.That(data.ResourceName).Key("data_disk.#").HasValue("0"),
+				check.That(data.ResourceName).Key("tags.%").HasValue("2"),
+				check.That(data.ResourceName).Key("tags.environment").HasValue("Dev"),
+				check.That(data.ResourceName).Key("tags.cost-center").HasValue("Ops"),
+			),
+		},
+	})
+}
+
+func (ImageDataSource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurestack" {
+  features {}
+}
+resource "azurestack_resource_group" "test" {
```

Comment:
could we add new lines to make this more readable?
```suggestion
}

resource "azurestack_resource_group" "test" {
```